### PR TITLE
fix: change restart policy to always

### PR DESCRIPTION
**Pull request by h4ck3rk3y**
DIVA node kills itself after an API call and needs to restart

### CODE REVIEW

File: container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go [hunk 1]
```diff
@@ -229,7 +229,7 @@ func (backend *DockerKurtosisBackend) StartRegisteredUserServices(ctx context.Co
 
 	var restartPolicy docker_manager.RestartPolicy = docker_manager.NoRestart
 	if backend.productionMode {
-		restartPolicy = docker_manager.RestartOnFailure
+		restartPolicy = docker_manager.RestartAlways
```
Comment (leoporoli) [inline container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go:232]: @h4ck3rk3y Do you know why RestartOnFailure did not restart the container in the DIVA use case?

Comment (h4ck3rk3y) [reply container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go:232]: Yes, it didn't restart for DIVA. I built this image and restarted and DIVA was restartign!

Comment (leoporoli) [reply container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go:232]: shouldn't they refactor their code to exit >0 if their service dies?

Comment (h4ck3rk3y) [reply container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go:232]: So the service dying is the expected behavior in this case! There's an API call that you make and that API call is meant to restart the service but I empathize

Comment (leoporoli) [reply container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go:232]: but probably yes, it's better to always restart a service in production mode

**Review by leoporoli — APPROVED** [review]

**Pull request merged**